CMakeLists.txt

```diff
@@ -104,7 +104,7 @@ option(Kokkos_ENABLE_DEPRECATION_WARNINGS "Enable Kokkos deprecation warnings" O
 include(FetchContent)
 FetchContent_Declare(kokkos
   GIT_REPOSITORY https://github.com/kokkos/kokkos.git
-  GIT_TAG        3.6.00
+  GIT_TAG        3.7.01
```

Comment:
Did we decide to update to the latest Kokkos release?
Does does coincide with the same version used to build Lightning?

Reply:
Lightning is now `3.7.0` now and let me make a change to match.